boot/cmdline.go

```diff
@@ -169,8 +169,25 @@ func composeCommandLine(model *asserts.Model, currentOrCandidate int, mode, syst
 		}
 		return "", err
 	}
-	// TODO:UC20: fetch extra args from gadget
 	extraArgs := ""
+	if gadgetDirOrSnapPath != "" {
+		sf, err := snapfile.Open(gadgetDirOrSnapPath)
```

Comment:
shouldn't we merge this Open into gadget.KernelCommandLineFromGadget ? do we have a case where we have the container already?

Reply:
Yes, that's a good point. I'll do it in a followup so that #10134  is unblocked first, if that's ok with you.